pkg/controller/template/render.go

```diff
@@ -47,6 +47,13 @@ const (
 //                                    /files/hostname.tmpl
 //
 func generateMachineConfigs(config *renderConfig, templateDir string) ([]*mcfgv1.MachineConfig, error) {
+	var tempIgnConfig ignv2_2types.Config
```

Comment:
this piece of code assumes only 2 roles worker,master which conflicts with the other part of the code that discovers the roles based on filestructure...

Reply:
In progress.

Reply:
@abhinavdahiya PTAL at new commit with test [here](https://github.com/openshift/machine-config-operator/pull/164/commits/9527f23128fa3ba4016b6f9c7b7f10f602fc6e9d)